test/UriTest.php

```diff
@@ -88,7 +89,7 @@ public function testWithPortReturnsNewInstanceWithProvidedPort($port)
         $this->assertNotSame($uri, $new);
         $this->assertEquals($port, $new->getPort());
         $this->assertEquals(
-            sprintf('https://user:pass@local.example.com:%d/foo?bar=baz#quz', $port),
+            sprintf('https://user:pass@local.example.com%s/foo?bar=baz#quz', $port === null ? '' : ':'.$port),
```

Comment:
Its preferible provide the expected values as part of the data provider. What test verify this branch of logic test as expected?